.travis.yml

```diff
@@ -10,6 +11,13 @@ env:
   # CF_PASSWORD
   - secure: UYvRRq3/WSAF5udIroJRoN0i4ha04K9tWf5ozXAidP9oF1o3kAdz7dMlb/UWE10KCki8WwYSKufsJknSC7yygg6BgkajpUFWZc36XWUgyIlXEti3kxjq8GhHUtVQpdH/Mp702DgzJCgA4Bwj9TlfkyP+PIYW4PfljRllJtPzqgxSur++23Q+kMmvA8T/GOEyab72ZjEMQmonl0Vxf6UWx/y7/4+XLj34OTzoYQ18utWfH9o9i1KUA8dYFCT3oCauGXF8Ra6iOPmNhBjvKrT9+foyYvfbwtL+o+tcbyBQM+p2toWUC2E5e+gIyed+woNnUMFGjGzkwzRe5evhH7RbssGMnkAHn19EGht+Ycdo5Wqh06kmb6sGVFa5EfvuX8AUHaOJrjBjS4ojHA9rmrkCrLvXpKECJi/NK2if5mk39ULtmtUzBBAjgPY3ZY34wweKrRgKz1Q2+Z59nJuX4/UM/KP3lrG0IWKIqdcrnoRaXsWADmf512zFw699+3rre6Bv7h08cyJZWAPWFhZiFibFOQXpT7uVeGyF2VcNrDjVzOiUFOWb7+5IRLi3Volp9o5OHuH1aOw3vNb2KqbIvJAFcQJF4l7c8Jopp6obeyBtXsupAVOTKZQnc9rokwDvyLbCNfseXXWNPekRD9+Ey8asSM+NauWLASFIWxNrRj6M5m0=
 
+before_install:
+  - echo "//registry.npmjs.org/:_authToken=\${NPM_TOKEN}" > .npmrc
```

Comment:
Where is NPM_TOKEN coming from now? Is it being set in the Heroku interface?

Reply:
Yes, it's being set in the Heroku interface

Reply:
I think it's strange to set environment variables in two different ways, so I'd like to choose one or the other. I think I prefer to set them both in this file - especially as we do it this way across most of the repos we own – but what made you choose the interface over this?

Reply:
went with this https://docs.travis-ci.com/user/private-dependencies/#API-Token and https://remysharp.com/2015/10/26/using-travis-with-private-npm-deps

if i use travis encrypt it will override current - secure (which is for cf) so not sure if it adds to it or flat out replaces it

Reply:
It would replace it – you'd need to encrypt both the `CF_PASSWORD` environment variable and the `NPM_TOKEN` at the same time. Which is one of the reasons why the comment above it is important.

Reply:
ok, where is cf password stored, so i can encrypt is?